android/app/src/main/java/org/uwaterloo/subletr/infrastructure/SubletrViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package org.uwaterloo.subletr.infrastructure
+
+import androidx.lifecycle.ViewModel
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.disposables.Disposable
+
+open class SubletrViewModel<UiState : Any>(
```

Comment:
This is good for now, but in the future it would be nice to strongly type this along with the children of the SubletrChildViewModel

Reply:
This does strongly type the `UiState` type.
All `: Any` means is that the type `UiState` has to be a subclass of `Any`.

I guess we could have stricter enforcement of typing of `SubletrChildViewModel`s if that's what you're saying.

Reply:
Nope, having a subclass of UiState works. Didn't know about those semantics, sorry.

For the childViewModels, yeah but not important right now.

Reply:
Nah, Kotlin syntax is just weird (I still think so despite having used the language for years).